src/main/java/com/sap/dsc/aas/lib/ua/transform/BrowsepathXPathBuilder.java

```diff
@@ -0,0 +1,228 @@
+package com.sap.dsc.aas.lib.ua.transform;
+
+import com.sap.dsc.aas.lib.transform.XPathBuilder;
+import com.sap.dsc.aas.lib.transform.XPathHelper;
+import org.dom4j.Document;
+import org.dom4j.Element;
+import org.dom4j.Node;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.invoke.MethodHandles;
+import java.util.*;
+import java.util.stream.Collectors;
+
+//FIXME the purpose is to check feasibility of getting XPath expression for NodeId from browsepath
+// it was originally written using W3C library and then be switched directly to dom4j, so it needs a lot of improvements and optimization
```

Comment:
either remove this comment or document the required improvements or optimizations